src/App.tsx

```diff
@@ -1,61 +1,109 @@
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const allTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const calcMaxTotoId = Math.max(0, ...allTodos.map(({ id }) => id)) + 1;
```

Comment:
this value calculates just one time. If you create several todos, ids will be the same (so will have several elements with the same keys)